speech/unit_tests/test_client.py

```diff
@@ -237,25 +240,39 @@ def test_sync_recognize_with_gax(self):
         from google.cloud._testing import _Monkey
 
         from google.cloud import speech
-        from google.cloud.speech import _gax as MUT
+        from google.cloud.speech import _gax
 
         creds = _Credentials()
         client = self._makeOne(credentials=creds, use_gax=True)
         client.connection = _Connection()
         client.connection.credentials = creds
         client._speech_api = None
-        transcript = 'testing 1 2 3'
-        confidence = 0.9224355
-        result = self._make_result(transcript, confidence)
-        _MockGAPICSpeechAPI._response = self._make_sync_response(result)
-
-        with _Monkey(MUT, SpeechApi=_MockGAPICSpeechAPI):
+        alternatives = [{
+            "transcript": 'testing 1 2 3',
+            "confidence": 0.9224355
+        }, {
+            "transcript": 'testing 4 5 6',
+            "confidence": 0.0123456
+        }]
+        result = self._make_result(alternatives)
+
+        def speech_api():
+            return _MockGAPICSpeechAPI(
+                response=self._make_sync_response(result))
+
+        with _Monkey(_gax, SpeechApi=speech_api):
             sample = client.sample(source_uri=self.AUDIO_SOURCE_URI,
                                    encoding=speech.Encoding.FLAC,
                                    sample_rate=self.SAMPLE_RATE)
             results = client.sync_recognize(sample)
-            self.assertEqual(results[0].transcript, transcript)
-            self.assertEqual(results[0].confidence, confidence)
+            self.assertEqual(results[0].transcript,
+                             alternatives[0]['transcript'])
+            self.assertEqual(results[0].confidence,
+                             alternatives[0]['confidence'])
+            self.assertEqual(results[1].transcript,
+                             alternatives[1]['transcript'])
+            self.assertEqual(results[1].confidence,
+                             alternatives[1]['confidence'])
```

Comment:
1. Don't create the `sample` in the context manager or do the asserts in the context manager.
1. Assert `len(results) == 2`
1. Do the same monkey-patch on the `client` object suggested above.